### PR TITLE
Modify internal base image struct

### PR DESCRIPTION
**Pull request by marcusmartins**
Modify the internal structure on how we store our base image compatibility matrix.

* Instead of hardcoding base images, we use the latest _n_ versions of CUDA, Python, and Torch from the Torch compatibility matrix.
* Patch versions are stripped out.

### CODE REVIEW

File: pkg/cli/baseimage.go [hunk 1]
```diff
@@ -44,17 +44,37 @@ func NewBaseImageRootCommand() (*cobra.Command, error) {
...
+		newVersionsMatrix(),
 	)
 
 	return &rootCmd, nil
 }
 
+func newVersionsMatrix() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "generate-versions",
+		Short: "Generate a nested matrix of python, cuda, pytorch versions (json)",
```
[the remark this answers is not in the document]
Comment (marcusmartins) [reply pkg/cli/baseimage.go:56]: In the future, we should add tensorflow images but now we are optimizing for images that we see more frequently. Cog uses can continue to build images with tensorflow, we will just not have pre-built images for now.

File: pkg/dockerfile/base.go [hunk 2]
```diff
@@ -40,6 +40,24 @@ var (
...
+type CUDAVersion struct {
+	Version string `json:"versions"`
+}
+
+type PyTorchVersion struct {
+	Version string `json:"version"`
+}
+
+type PythonVersion struct {
+	Version string           `json:"version"`
```
[the remark this answers is not in the document]
Comment (marcusmartins) [reply pkg/dockerfile/base.go:52]: I was trying to reflect the fact that python is the only common component across all versions. IMO having python at the top level key makes that easy to visualize and understand the matrix.

As I learn more about the compatibility package, I think we should leverage the version depedency from that package and keep the base image mapping simpler

File: pkg/cli/baseimage.go [hunk 3]
```diff
@@ -44,15 +44,34 @@ func NewBaseImageRootCommand() (*cobra.Command, error) {
...
+				return err
+			}
+			fmt.Println(string(output))
+			return nil
+		},
+		Args: cobra.MaximumNArgs(0),
+	}
+	return cmd
+}
+
```
Comment (mattt) [inline pkg/cli/baseimage.go:70]: What's the difference between this new command and the existing `generate-matrix` command implemented below?

Comment (mattt) [reply pkg/cli/baseimage.go]: As far as I can tell, `generate-versions` was an unused duplicate, so I removed it with 6d6d54c4ed0e018418d382795a8ba24947cbac24

**Review by mattt — APPROVED** [review]

**Pull request merged**